src/__experimental__/components/numeral-date/numeral-date.component.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import invariant from 'invariant';
+import Events from '../../../utils/helpers/events';
+import { StyledNumeralDate, StyledDateField } from './numeral-date.style';
+import Textbox from '../textbox';
+
+const NumeralDate = ({
+  dateFormat,
+  defaultValue,
+  errorMessage,
+  errorPresent,
+  id,
+  name,
+  onBlur,
+  onChange,
+  value
+}) => {
+  const isControlled = value !== undefined;
+  const initialValue = isControlled ? value : defaultValue;
+  invariant(
+    initialValue !== undefined,
+    'This component has no initial value'
+  );
+
+  const [isActive, setIsActive] = useState();
+  const [dateValue, setDateValue] = useState({
+    ...initialValue
+  });
+
+  const onKeyPress = (ev) => {
+    const isValidKey = Events.isNumberKey(ev) || Events.isDeletingKey(ev) || Events.isTabKey(ev);
+
+    if (!isValidKey) {
+      ev.preventDefault();
+    }
+  };
+
+  const handleChange = (e, itemId) => {
+    if (onChange) {
+      onChange(e);
+    }
+    if (e.target.value !== dateValue[itemId]) {
+      setDateValue({ ...dateValue, [itemId]: e.target.value });
+    }
+  };
+
+  const handleOnFocus = () => {
+    setIsActive(true);
+  };
+
+  const handleBlur = (ev) => {
+    const targetObject = {
+      name: ev.target.name,
+      id: ev.target.id,
+      value: { ...dateValue }
+    };
+
+    setIsActive(false);
+
+    if (onBlur) {
+      onBlur(targetObject);
+    }
+  };
+
+  return (
+    <StyledNumeralDate
+      name={ name }
+      id={ id }
+      isActive={ isActive }
+      onBlur={ handleBlur }
+      onKeyPress={ onKeyPress }
+      onFocus={ handleOnFocus }
+      data-component='numeral-date'
+    >
+      {
+        dateFormat.map((datePart, textboxNumber) => {
+          const isEnd = textboxNumber === dateFormat.length - 1;
+          return (
+            <StyledDateField
+              key={ datePart }
+              isYearInput={ datePart.length === 4 }
+              isMiddle={ textboxNumber === 1 }
+              isEnd={ isEnd }
+              twoPartDate={ textboxNumber <= 1 }
+              errorPresent={ errorPresent }
+              dateFormatLength={ dateFormat.length }
+            >
+              <Textbox
+                placeholder={ datePart }
+                value={ dateValue[datePart] }
+                onChange={ e => handleChange(e, datePart) }
+                hasError={ errorPresent }
+                onBlur={ handleBlur }
+                {
+                ...(isEnd && errorPresent && {
```

Comment:
I think it would be best if `NumeralDate` would accept following interface for validations:
`{ error: 'string', warning: 'string', info: 'string' }`

Internally in `NumeralDate` these values have to be transformed into an interface that'a currently accepted by `Textbox`
This should more or less look like that: 
```js
{
  ...(isEnd && error && {
    inputIcon: 'error',
    hasError: 'true',
    tooltipMessage: error
  }
}
```

Reply:
This has been addressed in commit 3a416145924954235da8036ea2fc2000b6375dd8. Thank you.